openpype/modules/deadline/plugins/publish/submit_publish_job.py

```diff
@@ -192,8 +191,7 @@ def _submit_deadline_post_job(self, instance, job, instances):
 
         # Transfer the environment from the original job to this dependent
         # job so they use the same environment
-        metadata_path, rootless_metadata_path = \
-            create_metadata_path(instance, anatomy)
+        _, rootless_metadata_path = create_metadata_path(instance, anatomy)
```

Comment:
in our fork of OP I have changed the `_submit_deadline_post_job` function so it not only returns `deadline_publish_job_id` but also the `metadata_path` (the one you are now ignoring with `_`) to avoid the second call of this same function later on the code path here https://github.com/ynput/OpenPype/blob/eec21037a3c79b000afea44edd1494af397f3c14/openpype/modules/deadline/plugins/publish/submit_publish_job.py#L508

So it ends looking like this:
```python
deadline_publish_job_id, metadata_path = self._submit_deadline_post_job(instance, render_job, instances)
```

The reason we had to do this is because in our Hiero integration we are submitting farm jobs for different assets and subsets from the same parent "asset" (so we can ingest multiple shots from just the episode context) and so I changed the metadata.json path to be unique for each subset and it also helps to add a timestamp for debugging purposes and enforcing even more uniqueness:
```
    # Prefixing metadata file with timestamp and asset so the .json files are
    # unique and not overwrite each other. This is necessary because in Hiero
    # we use the same working directory to publish multiple subsets at once
    # and when the subset was called the same, it was overwriting the same file
    # over and over
    metadata_filename = "{}_{}_{}_metadata.json".format(
        datetime.datetime.now().strftime("%d%m%Y%H%M%S"),
        ins_data["asset"],
        ins_data["subset"]
    )
```
And because of the current logic of calling the `create_metadata_path` twice... that timestamp would return a second difference between both executions,

Making the json files unique might not be something you are interested in but I think it makes the `submit_publish_job` code more understandable too if you only "create" that metadata path once (it was confusing for me to read the first time why there were two "creations" of the path)

** Bonus points: I think `get_metadata_path` defines semantically a bit better what the function does as the file isn't being created in the function

Reply:
@jakubjezek001 have you seen this?

Reply:
@fabiaserra I totally agree this should be redesigned with your suggestion. The plan is to include the editorial publishing at the farm too, so this would need to be dealt with anyway. What do  you thing @iLLiCiTiT - should this be part of this PR?